config/jsonconfig/config.go

```diff
@@ -0,0 +1,109 @@
+/*
+jsonconfig implements configuration, reading json into an ice Module.
+
+To use:
+
+1) Create the Options. List your configurable Interfaces. Each Interface
+can be backed by several named Implementations.
+2) Options.Parse parses bytes and creates a Configuration.
+  a) for each Interface, pick which Implementation.
+  b) json.Unmarshal the json into that Implementation
+  c) Implementation can now be used as a Module or json.Marshal'ed to print its configuration
+3) Configuration is an ice Module that installs all the configured Implementations
+*/
+package jsonconfig
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"github.com/scootdev/scoot/ice"
+)
+
+type Options map[string]Interface
+
+type Interface map[string]Implementation
```

Comment:
please don't use reserved words as object types this is super confusing.  I don't have any idea what this does